google/resource_bigtable_table.go

```diff
@@ -0,0 +1,137 @@
+package google
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"golang.org/x/net/context"
+	"log"
+)
+
+func resourceBigtableTable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceBigtableTableCreate,
+		Read:   resourceBigtableTableRead,
+		Delete: resourceBigtableTableDestroy,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"instance_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"split_keys": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceBigtableTableCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	ctx := context.Background()
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	instanceName := d.Get("instance_name").(string)
+	c, err := config.bigtableClientFactory.NewAdminClient(project, instanceName)
+	if err != nil {
+		return fmt.Errorf("Error starting admin client. %s", err)
+	}
+
+	defer c.Close()
+
+	name := d.Get("name").(string)
+	if v, ok := d.GetOk("split_keys"); ok {
+		splitKeys := convertSchemaArrayToStringArray(v.([]interface{}))
+		err = c.CreatePresplitTable(ctx, name, splitKeys)
```

Comment:
For this and below- the docs say "This method may return before the table's creation is complete." They don't seem to be currently providing us good ways to check on its creation, but maybe add a note here that in the future we might need to loop and wait until the table exists